infra/ci/worker/gce-startup-script.sh

```diff
@@ -15,105 +15,37 @@
+shopt -s nullglob
 for SSD in /dev/nvme0n*; do
 mkswap $SSD
 swapon -p -1 $SSD
 done
 
-# This is used by the sandbox containers, NOT needed by the workers.
-# Rationale for size=500G: by default tmpfs mount are set to RAM/2, which makes
-# the CI depend too much on the underlying VM. Here and below, we pick an
-# arbitrary fixed size (we use local scratch NVME as a swap device).
-export SHARED_WORKER_CACHE=/mnt/disks/shared_worker_cache
-rm -rf $SHARED_WORKER_CACHE
-mkdir -p $SHARED_WORKER_CACHE
-mount -t tmpfs tmpfs $SHARED_WORKER_CACHE -o mode=777,size=500G
-
-# This is used to queue build artifacts that are uploaded to GCS.
-export ARTIFACTS_DIR=/mnt/disks/artifacts
-rm -rf $ARTIFACTS_DIR
-mkdir -p $ARTIFACTS_DIR
-mount -t tmpfs tmpfs $ARTIFACTS_DIR -o mode=777,size=500G
-
-# Pull the latest images from the registry.
-docker pull $WORKER_IMG
-docker pull $SANDBOX_IMG
+# Enlarge the /tmp size to use swap and also allow exec, as some tests need it.
+mount -o remount,mode=777,size=95%,exec /tmp
 
-# Create the restricted bridge for the sandbox container.
-# Prevent access to the metadata server and impersonation of service accounts.
-docker network rm sandbox 2>/dev/null || true  # Handles the reboot case.
-docker network create sandbox -o com.docker.network.bridge.name=sandbox
-sudo iptables -I DOCKER-USER -i sandbox -d 169.254.0.0/16 -j REJECT
+# Disable Addressa space ranomization. This is needed for Tsan tests to work in
+# the sandbox. Newer kernel versions seem to block the
+# personality(ADDR_NO_RANDOMIZE) call within the sandbox and would require the
+# sandbox to be --privileged, which defeats the point of having a sandbox.
+sudo sysctl -w kernel.randomize_va_space=0
 
-# These args will be appended to the docker run invocation for the sandbox.
-export SANDBOX_NETWORK_ARGS="--network sandbox --dns 8.8.8.8"
-
-# The worker_main_loop.py script creates one docker sandbox container for
-# each job invocation. It needs to talk back to the host docker to do so.
-# This implies that the worker container is trusted and should never run code
-# from the repo, as opposite to the sandbox container that is isolated.
-for i in $(seq $NUM_WORKERS); do
-
-# We manually mount a tmpfs mount ourselves because Docker doesn't allow to
-# both override tmpfs-size AND "-o exec" (see also
-# https://github.com/moby/moby/issues/32131)
-SANDBOX_TMP=/mnt/disks/sandbox-$i-tmp
-rm -rf $SANDBOX_TMP
-mkdir -p $SANDBOX_TMP
-mount -t tmpfs tmpfs $SANDBOX_TMP -o mode=777,size=100G
-
-docker rm -f worker-$i 2>/dev/null || true
 docker run -d \
-  -v /var/run/docker.sock:/var/run/docker.sock \
-  -v $ARTIFACTS_DIR:$ARTIFACTS_DIR \
-  --env SHARED_WORKER_CACHE="$SHARED_WORKER_CACHE" \
-  --env SANDBOX_NETWORK_ARGS="$SANDBOX_NETWORK_ARGS" \
-  --env ARTIFACTS_DIR="$ARTIFACTS_DIR" \
-  --env SANDBOX_TMP="$SANDBOX_TMP" \
-  --env WORKER_HOST="$(hostname)" \
-  --name worker-$i \
-  --hostname worker-$i \
+  --name worker \
+  --privileged \
```

Comment:
Do we need this? Seems like this contradicts the comment above about not wanting this.

Reply:
you mean the --privilege?
yeah the worker (i would rename it, but it's a pain) is really the root image. it's a docker container purely to make deployment and testing easy, but it's by all itnent and purposes as privileged as GCE.
it's just a way to make it easier to update the image (managing a raw linux image on GCE is a majestic pain)